diff --git a/main/util-linux/APKBUILD b/main/util-linux/APKBUILD
index 74d5a5e..8e73b27 100644
--- a/main/util-linux/APKBUILD
+++ b/main/util-linux/APKBUILD
@@ -111,7 +111,7 @@ libmount() {
 	pkgdesc="Block device identification library from util-linux"
 	depends=""
 	mkdir -p "$subpkgdir"/lib
-	mv "$pkgdir"/lib/libmount.so.* "$subpkgdir"/lib/
+	mv "$pkgdir"/lib/libmount* "$subpkgdir"/lib/
 }
 
 libsmartcols() {
