diff --git a/main/file/APKBUILD b/main/file/APKBUILD
index fbdb5bb..8a1935c 100644
--- a/main/file/APKBUILD
+++ b/main/file/APKBUILD
@@ -29,6 +29,7 @@ build() {
 		--host=$CHOST \
 		--prefix=/usr \
 		--datadir=/usr/share \
+		--enable-static \
 		|| return 1
 	make || return 1
 	make tests || return 1
diff --git a/main/gettext/APKBUILD b/main/gettext/APKBUILD
index 2808e6e..c6ec7c5 100644
--- a/main/gettext/APKBUILD
+++ b/main/gettext/APKBUILD
@@ -28,7 +28,7 @@ build() {
 		--prefix=/usr \
 		--enable-threads=posix \
 		--disable-java \
-		--disable-static \
+		--enable-static \
 		|| return 1
 	make || return 1
 }
diff --git a/main/glib/APKBUILD b/main/glib/APKBUILD
index 363005c..fda68c1 100644
--- a/main/glib/APKBUILD
+++ b/main/glib/APKBUILD
@@ -41,6 +41,7 @@ build() {
 		--disable-compile-warnings \
 		--with-pcre=system \
 		--with-pic \
+		--enable-static \
 		|| return 1
 	make || return 1
 }
diff --git a/main/libevent/APKBUILD b/main/libevent/APKBUILD
index 863bb0d..36df993 100644
--- a/main/libevent/APKBUILD
+++ b/main/libevent/APKBUILD
@@ -38,7 +38,7 @@ build() {
 		--host=$CHOST \
 		--prefix=/usr \
 		--sysconfdir=/etc \
-		--disable-static \
+		--enable-static \
 		|| return 1
 	make || return 1
 }
diff --git a/main/musl/APKBUILD b/main/musl/APKBUILD
index a311a2b..dde784a 100644
--- a/main/musl/APKBUILD
+++ b/main/musl/APKBUILD
@@ -115,6 +115,7 @@ package() {
 
 	# make LDSO the be the real file, and libc the symlink
 	local LDSO=$(make -f Makefile --eval "$(echo -e 'print-ldso:\n\t@echo $$(basename $(LDSO_PATHNAME))')" print-ldso)
+	mv -f "$pkgdir"/usr/lib/libc.a "$pkgdir"/lib/libc.a || return 1
 	mv -f "$pkgdir"/usr/lib/libc.so "$pkgdir"/lib/"$LDSO" || return 1
 	ln -sf "$LDSO" "$pkgdir"/lib/libc.musl-${CARCH}.so.1 || return 1
 	ln -sf ../../lib/"$LDSO" "$pkgdir"/usr/lib/libc.so || return 1
diff --git a/main/sqlite/APKBUILD b/main/sqlite/APKBUILD
index bea7686..a7a5181 100644
--- a/main/sqlite/APKBUILD
+++ b/main/sqlite/APKBUILD
@@ -48,7 +48,7 @@ build() {
 		--host=$CHOST \
 		--prefix=/usr \
 		--enable-threadsafe \
-		--disable-static \
+		--enable-static \
 		--enable-readline \
 		--enable-dynamic-extensions \
 		|| return 1
diff --git a/main/util-linux/APKBUILD b/main/util-linux/APKBUILD
index 3f75880..67deda5 100644
--- a/main/util-linux/APKBUILD
+++ b/main/util-linux/APKBUILD
@@ -117,7 +117,7 @@ libmount() {
 	pkgdesc="Block device identification library from util-linux"
 	depends=
 	mkdir -p "$subpkgdir"/lib
-	mv "$pkgdir"/lib/libmount.so.* "$subpkgdir"/lib/
+	mv "$pkgdir"/lib/libmount* "$subpkgdir"/lib/
 }
 
 libsmartcols() {
