diff --git a/main/musl/APKBUILD b/main/musl/APKBUILD
index ee790a9..6db8ff7 100644
--- a/main/musl/APKBUILD
+++ b/main/musl/APKBUILD
@@ -163,6 +163,7 @@ compat() {
 	esac
 	ln -sf "/lib/libc.musl-${CARCH}.so.1" "$subpkgdir/lib/$_ld"
 
+	mv -f "$pkgdir"/usr/lib/libc.a "$pkgdir"/lib/libc.a || return 1
 	for i in libc.so.6 libcrypt.so.1 libm.so.6 libpthread.so.0 librt.so.1 libutil.so.1; do
 		ln -sf "/lib/libc.musl-${CARCH}.so.1" "$subpkgdir/lib/$i"
 	done
