diff --git a/testing/static-rspamd/APKBUILD b/testing/static-rspamd/APKBUILD
new file mode 100644
index 0000000..97ca692
--- /dev/null
+++ b/testing/static-rspamd/APKBUILD
@@ -0,0 +1,83 @@
+# Maintainer: Andrew Lewis <nerf@judo.za.org>
+# Contributor: Valery Kartel <valery.kartel@gmail.com>
+pkgname=rspamd
+pkgver=1.5.7
+pkgrel=1
+pkgdesc="Rapid spam filtering system"
+url="https://rspamd.com"
+arch="all"
+license="BSD"
+pkgusers="rspamd"
+pkggroups="rspamd"
+depends=""
+depends_dev=""
+makedepends="$depends_dev cmake libressl-dev libevent-dev glib-dev
+	luajit-dev sqlite-dev file-dev pcre2-dev"
+options="!strip"
+install="$pkgname.pre-install"
+source="https://aluminium.tinfoilcat.org/rspamd-download/rspamd-1.5.7.tar.gz
+	$pkgname.logrotated
+	$pkgname.initd
+	$pkgname.confd
+	pcre.patch
+"
+
+_builddir="$srcdir"/$pkgname-$pkgver
+build() {
+	cd "$_builddir"
+	sed -i CMakeLists.txt -e "/LIST(APPEND CMAKE_REQUIRED_LIBRARIES nsl)/d"
+	cmake CMakeLists.txt \
+		-DCMAKE_INSTALL_PREFIX=/usr \
+		-DCONFDIR=/etc/$pkgname \
+		-DRUNDIR=/run/$pkgname \
+		-DRSPAMD_USER=$pkgusers \
+		-DRSPAMD_GROUP=$pkggroups \
+		-DENABLE_LUAJIT=ON \
+		-DENABLE_STATIC=ON \
+		-DENABLE_PCRE2=ON \
+		-DENABLE_HYPERSCAN=OFF \
+		-DHYPERSCAN_ROOT_DIR=/home/abuild/hyperscan \
+		-DENABLE_FANN=ON \
+		-DFANN_ROOT_DIR=/home/abuild/fann \
+		-DENABLE_HIREDIS=ON \
+		-DENABLE_GD=OFF \
+		-DINSTALL_EXAMPLES=ON \
+		|| return 1
+	make || return 1
+}
+
+package() {
+	cd "$_builddir"
+	make DESTDIR="$pkgdir" install || return 1
+	find "$pkgdir"/usr/bin -type l -delete
+
+	mkdir -p "$pkgdir"/etc/$pkgname/local.d "$pkgdir"/etc/$pkgname/override.d
+
+	install -Dm644 "$srcdir"/$pkgname.logrotated "$pkgdir"/etc/logrotate.d/$pkgname
+	install -Dm755 "$srcdir"/$pkgname.initd "$pkgdir"/etc/init.d/$pkgname
+	install -Dm644 "$srcdir"/$pkgname.confd "$pkgdir"/etc/conf.d/$pkgname
+
+	install -dm750 -g $pkggroups "$pkgdir"/var/log/$pkgname
+
+	mkdir "$pkgdir"/usr/sbin
+	mv "$pkgdir"/usr/bin/rspamd-$pkgver "$pkgdir"/usr/sbin/rspamd
+	mv "$pkgdir"/usr/bin/rspamadm-$pkgver "$pkgdir"/usr/bin/rspamadm
+	mv "$pkgdir"/usr/bin/rspamc-$pkgver "$pkgdir"/usr/bin/rspamc
+	rm -rf "$pkgdir"/usr/share/man
+}
+
+md5sums="4f54faf12643fb900dca096b41294e9a  rspamd-1.5.7.tar.gz
+c152c6a90f6ae9e5a7a1d137dfbc0305  rspamd.logrotated
+3f32a16e76e1461ffba9231cfb1e3d17  rspamd.initd
+ecfea2a25b95727ae91c04001fbd3e46  rspamd.confd
+eb96a453da1fce142542d5174ff22748  pcre.patch"
+sha256sums="73d18b129a072e35c74c9a440abdd0985fa15696d308a5e23736ec007432fd70  rspamd-1.5.7.tar.gz
+6c5e79e9052d957f3d0d634b2ae7a56bbc0901a5d6946dc991c92f19a72fce97  rspamd.logrotated
+6b531f95724b2a3990524ab09b7304ce4e811b6e082dfdbe633f201a6bc7eee3  rspamd.initd
+82be6a663af2e2333b0dfbbbfd05a9ff3d02e05c7e506235b1b0dbd9d0b72972  rspamd.confd
+73c845e87345a9c85ccb7dd5495d123ffea28cf36d2cb8cf5312979be794f19e  pcre.patch"
+sha512sums="362f8615795c739d1c869dd48426dadee38924287d82ad725baf0d3f80aa4571db29501d95d1497d3dfef61af39c2c7254f2c9437ee46f453e380f595b5873ca  rspamd-1.5.7.tar.gz
+2efe28575c40d1fba84b189bb872860e744400db80dce2f6330be6c6287fb3f46e6511284729b957488bf40bcb9b0952e26df9934f5f138334bd2766075c45cb  rspamd.logrotated
+30b45812ef68f2b82d0d7f370b44bec52691296c7349c96c8273342eb4f9b5708c13ad97b13f63d81bee588b4e459c0da3092a62adff9e5b8938f44546df3dcd  rspamd.initd
+0b73b159cec9a4a1d337fbb429814f78da23b55f72c9fb8a777ab5f06634206a4f9b25e587f8dbfa7c3242ac5501ebcc90b9a0e926adfd37e14a12ac4607fa62  rspamd.confd
+342e989b172fe8fddffa1a1d09c9604f7e326ca2ec9e104d04451c363c73942756508f03a6d13fef5d3649bcb87863ce29a5f8a6d871a2dc7086fa7b9e787b19  pcre.patch"
diff --git a/testing/static-rspamd/pcre.patch b/testing/static-rspamd/pcre.patch
new file mode 100644
index 0000000..c9dc77d
--- /dev/null
+++ b/testing/static-rspamd/pcre.patch
@@ -0,0 +1,13 @@
+diff --git a/src/libutil/map.c b/src/libutil/map.c
+index 57c84e3..97a9827 100644
+--- a/src/libutil/map.c
++++ b/src/libutil/map.c
+@@ -2290,7 +2290,7 @@ rspamd_re_map_insert_helper (gpointer st, gconstpointer key, gconstpointer value
+ 	}
+ #else
+ 	if (pcre_flags & PCRE_FLAG(UTF)) {
+-		re_map->map_flags |= RSPAMD_REGEXP_FLAG_UTF;
++		re_map->has_utf = TRUE;
+ 	}
+ #endif
+ 
diff --git a/testing/static-rspamd/rspamd.confd b/testing/static-rspamd/rspamd.confd
new file mode 100644
index 0000000..541ffff
--- /dev/null
+++ b/testing/static-rspamd/rspamd.confd
@@ -0,0 +1,17 @@
+# User to run rspamd workers
+#user=rspamd
+
+# Group to run rspamd workers
+#group=rspamd
+
+# Main config file
+#cfgfile=/etc/rspamd/rspamd.conf
+
+# PID file. Settings from config files are ignored
+#pidfile=/run/rspamd/rspamd.pid
+
+# Where to log startup configuration checking
+# - /dev/null - silent (default)
+# - /dev/tty - show output on curent terminal
+# - /path/filename - append to specified logfile
+#startuplog=/var/log/rspamd/startup.log
diff --git a/testing/static-rspamd/rspamd.initd b/testing/static-rspamd/rspamd.initd
new file mode 100644
index 0000000..a20f784
--- /dev/null
+++ b/testing/static-rspamd/rspamd.initd
@@ -0,0 +1,37 @@
+#!/sbin/openrc-run
+
+description="Rapid spam filtering system"
+user=${user:-rspamd}
+group=${group:-rspamd}
+cfgfile=${cfgfile:-/etc/rspamd/rspamd.conf}
+pidfile=${pidfile:-/run/rspamd/rspamd.pid}
+command=/usr/sbin/rspamd
+command_args="-u $user -g $group -c $cfgfile -p $pidfile"
+required_files="$cfgfile"
+extra_started_commands="reload reopen"
+description_reload="Reload configuration"
+description_reopen="Reopen log files"
+
+depend() {
+	need localmount net
+	before exim postfix
+}
+
+start_pre() {
+	ebegin
+	checkpath --directory --mode 750 --owner $user:$group ${pidfile%/*}
+	$command $command_args -t >/dev/null 2>>${startuplog:-/dev/null}
+	eend $?
+}
+
+reload() {
+	ebegin "Reloading ${SVCNAME}"
+	start-stop-daemon --signal HUP --pidfile $pidfile
+	eend $?
+}
+
+reopen() {
+	ebegin "Reopening ${SVCNAME} log files"
+	start-stop-daemon --signal USR1 --pidfile $pidfile
+	eend $?
+}
\ No newline at end of file
diff --git a/testing/static-rspamd/rspamd.logrotated b/testing/static-rspamd/rspamd.logrotated
new file mode 100644
index 0000000..2a443f2
--- /dev/null
+++ b/testing/static-rspamd/rspamd.logrotated
@@ -0,0 +1,11 @@
+/var/log/rspamd/*log {
+    daily
+    rotate 10
+    missingok
+    notifempty
+    compress
+    sharedscripts
+    postrotate
+        /etc/init.d/rspamd --ifstarted --quiet reopen
+    endscript
+}
diff --git a/testing/static-rspamd/rspamd.pre-install b/testing/static-rspamd/rspamd.pre-install
new file mode 100644
index 0000000..b37dcf3
--- /dev/null
+++ b/testing/static-rspamd/rspamd.pre-install
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+addgroup -S rspamd 2>/dev/null
+adduser -S -D -H -h /var/lib/rspamd -s /bin/false -G rspamd -g rspamd rspamd 2>/dev/null
+
+exit 0
