diff --git a/main/musl/APKBUILD b/main/musl/APKBUILD
index a311a2b..dde784a 100644
--- a/main/musl/APKBUILD
+++ b/main/musl/APKBUILD
@@ -115,6 +115,7 @@ package() {
 
 	# make LDSO the be the real file, and libc the symlink
 	local LDSO=$(make -f Makefile --eval "$(echo -e 'print-ldso:\n\t@echo $$(basename $(LDSO_PATHNAME))')" print-ldso)
+	mv -f "$pkgdir"/usr/lib/libc.a "$pkgdir"/lib/libc.a || return 1
 	mv -f "$pkgdir"/usr/lib/libc.so "$pkgdir"/lib/"$LDSO" || return 1
 	ln -sf "$LDSO" "$pkgdir"/lib/libc.musl-${CARCH}.so.1 || return 1
 	ln -sf ../../lib/"$LDSO" "$pkgdir"/usr/lib/libc.so || return 1
