commit ce88625203dc95cbb42f49f1a1532a98f1182409
Author: Andrew Lewis <nerf@judo.za.org>
Date:   Mon May 29 12:09:42 2017 +0200

    rspamd

diff --git a/testing/static-rspamd/APKBUILD b/testing/static-rspamd/APKBUILD
new file mode 100644
index 0000000..c2775ee
--- /dev/null
+++ b/testing/static-rspamd/APKBUILD
@@ -0,0 +1,71 @@
+# Maintainer: Andrew Lewis <nerf@judo.za.org>
+# Contributor: Valery Kartel <valery.kartel@gmail.com>
+pkgname=rspamd
+pkgver=1.5.8
+pkgrel=1
+pkgdesc="Rapid spam filtering system"
+url="https://rspamd.com"
+arch="all"
+license="BSD"
+pkgusers="rspamd"
+pkggroups="rspamd"
+depends=""
+depends_dev=""
+makedepends="$depends_dev cmake libressl-dev libevent-dev glib-dev
+	luajit-dev sqlite-dev file-dev pcre2-dev"
+options="!strip"
+install="$pkgname.pre-install"
+source="https://aluminium.tinfoilcat.org/rspamd-snap/rspamd-1.5.8.tar.gz
+	$pkgname.logrotated
+	$pkgname.initd
+	$pkgname.confd
+"
+
+_builddir="$srcdir"/$pkgname-$pkgver
+build() {
+	cd "$_builddir"
+	sed -i CMakeLists.txt -e "/LIST(APPEND CMAKE_REQUIRED_LIBRARIES nsl)/d"
+	cmake CMakeLists.txt \
+		-DCMAKE_INSTALL_PREFIX=/usr \
+		-DCONFDIR=/etc/$pkgname \
+		-DRUNDIR=/run/$pkgname \
+		-DRSPAMD_USER=$pkgusers \
+		-DRSPAMD_GROUP=$pkggroups \
+		-DENABLE_LUAJIT=ON \
+		-DENABLE_STATIC=ON \
+		-DENABLE_PCRE2=ON \
+		-DENABLE_HYPERSCAN=ON \
+		-DHYPERSCAN_ROOT_DIR=/home/abuild/hyperscan \
+		-DENABLE_FANN=ON \
+		-DFANN_ROOT_DIR=/home/abuild/fann \
+		-DENABLE_HIREDIS=ON \
+		-DENABLE_GD=OFF \
+		-DINSTALL_EXAMPLES=ON \
+		|| return 1
+	make || return 1
+}
+
+package() {
+	cd "$_builddir"
+	make DESTDIR="$pkgdir" install || return 1
+	find "$pkgdir"/usr/bin -type l -delete
+
+	mkdir -p "$pkgdir"/etc/$pkgname/local.d "$pkgdir"/etc/$pkgname/override.d
+
+	install -Dm644 "$srcdir"/$pkgname.logrotated "$pkgdir"/etc/logrotate.d/$pkgname
+	install -Dm755 "$srcdir"/$pkgname.initd "$pkgdir"/etc/init.d/$pkgname
+	install -Dm644 "$srcdir"/$pkgname.confd "$pkgdir"/etc/conf.d/$pkgname
+
+	install -dm750 -g $pkggroups "$pkgdir"/var/log/$pkgname
+
+	mkdir "$pkgdir"/usr/sbin
+	mv "$pkgdir"/usr/bin/rspamd-$pkgver "$pkgdir"/usr/sbin/rspamd
+	mv "$pkgdir"/usr/bin/rspamadm-$pkgver "$pkgdir"/usr/bin/rspamadm
+	mv "$pkgdir"/usr/bin/rspamc-$pkgver "$pkgdir"/usr/bin/rspamc
+	rm -rf "$pkgdir"/usr/share/man
+}
+
+sha512sums="9ba18c15458a14673db62cd4db7d6bc54057f97d0e77ffca816d4fb2cee86c7deffed9261f05b33cd9261ec0a0939555874d800663cd7b78e9356cc653c2a4bf  rspamd-1.5.8.tar.gz
+2efe28575c40d1fba84b189bb872860e744400db80dce2f6330be6c6287fb3f46e6511284729b957488bf40bcb9b0952e26df9934f5f138334bd2766075c45cb  rspamd.logrotated
+30b45812ef68f2b82d0d7f370b44bec52691296c7349c96c8273342eb4f9b5708c13ad97b13f63d81bee588b4e459c0da3092a62adff9e5b8938f44546df3dcd  rspamd.initd
+0b73b159cec9a4a1d337fbb429814f78da23b55f72c9fb8a777ab5f06634206a4f9b25e587f8dbfa7c3242ac5501ebcc90b9a0e926adfd37e14a12ac4607fa62  rspamd.confd"
diff --git a/testing/static-rspamd/rspamd.confd b/testing/static-rspamd/rspamd.confd
new file mode 100644
index 0000000..541ffff
--- /dev/null
+++ b/testing/static-rspamd/rspamd.confd
@@ -0,0 +1,17 @@
+# User to run rspamd workers
+#user=rspamd
+
+# Group to run rspamd workers
+#group=rspamd
+
+# Main config file
+#cfgfile=/etc/rspamd/rspamd.conf
+
+# PID file. Settings from config files are ignored
+#pidfile=/run/rspamd/rspamd.pid
+
+# Where to log startup configuration checking
+# - /dev/null - silent (default)
+# - /dev/tty - show output on curent terminal
+# - /path/filename - append to specified logfile
+#startuplog=/var/log/rspamd/startup.log
diff --git a/testing/static-rspamd/rspamd.initd b/testing/static-rspamd/rspamd.initd
new file mode 100644
index 0000000..a20f784
--- /dev/null
+++ b/testing/static-rspamd/rspamd.initd
@@ -0,0 +1,37 @@
+#!/sbin/openrc-run
+
+description="Rapid spam filtering system"
+user=${user:-rspamd}
+group=${group:-rspamd}
+cfgfile=${cfgfile:-/etc/rspamd/rspamd.conf}
+pidfile=${pidfile:-/run/rspamd/rspamd.pid}
+command=/usr/sbin/rspamd
+command_args="-u $user -g $group -c $cfgfile -p $pidfile"
+required_files="$cfgfile"
+extra_started_commands="reload reopen"
+description_reload="Reload configuration"
+description_reopen="Reopen log files"
+
+depend() {
+	need localmount net
+	before exim postfix
+}
+
+start_pre() {
+	ebegin
+	checkpath --directory --mode 750 --owner $user:$group ${pidfile%/*}
+	$command $command_args -t >/dev/null 2>>${startuplog:-/dev/null}
+	eend $?
+}
+
+reload() {
+	ebegin "Reloading ${SVCNAME}"
+	start-stop-daemon --signal HUP --pidfile $pidfile
+	eend $?
+}
+
+reopen() {
+	ebegin "Reopening ${SVCNAME} log files"
+	start-stop-daemon --signal USR1 --pidfile $pidfile
+	eend $?
+}
\ No newline at end of file
diff --git a/testing/static-rspamd/rspamd.logrotated b/testing/static-rspamd/rspamd.logrotated
new file mode 100644
index 0000000..2a443f2
--- /dev/null
+++ b/testing/static-rspamd/rspamd.logrotated
@@ -0,0 +1,11 @@
+/var/log/rspamd/*log {
+    daily
+    rotate 10
+    missingok
+    notifempty
+    compress
+    sharedscripts
+    postrotate
+        /etc/init.d/rspamd --ifstarted --quiet reopen
+    endscript
+}
diff --git a/testing/static-rspamd/rspamd.pre-install b/testing/static-rspamd/rspamd.pre-install
new file mode 100644
index 0000000..b37dcf3
--- /dev/null
+++ b/testing/static-rspamd/rspamd.pre-install
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+addgroup -S rspamd 2>/dev/null
+adduser -S -D -H -h /var/lib/rspamd -s /bin/false -G rspamd -g rspamd rspamd 2>/dev/null
+
+exit 0
