diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index fade00c..a86719e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -11,7 +11,7 @@ SET(floatfann_LIB_SRCS
 floatfann.c
 )
 
-ADD_LIBRARY(floatfann SHARED ${floatfann_LIB_SRCS})
+ADD_LIBRARY(floatfann STATIC ${floatfann_LIB_SRCS})
 
 SET_TARGET_PROPERTIES(floatfann PROPERTIES VERSION ${VERSION} SOVERSION 2 )
 INSTALL(TARGETS floatfann DESTINATION ${LIB_INSTALL_DIR} )
@@ -23,7 +23,7 @@ SET(doublefann_LIB_SRCS
 doublefann.c
 )
 
-ADD_LIBRARY(doublefann SHARED ${doublefann_LIB_SRCS})
+ADD_LIBRARY(doublefann STATIC ${doublefann_LIB_SRCS})
 
 SET_TARGET_PROPERTIES(doublefann PROPERTIES VERSION ${VERSION} SOVERSION 2 )
 INSTALL(TARGETS doublefann DESTINATION ${LIB_INSTALL_DIR} )
@@ -35,7 +35,7 @@ SET(fixedfann_LIB_SRCS
 fixedfann.c
 )
 
-ADD_LIBRARY(fixedfann SHARED ${fixedfann_LIB_SRCS})
+ADD_LIBRARY(fixedfann STATIC ${fixedfann_LIB_SRCS})
 
 SET_TARGET_PROPERTIES(fixedfann PROPERTIES VERSION ${VERSION} SOVERSION 2 )
 INSTALL(TARGETS fixedfann DESTINATION ${LIB_INSTALL_DIR} )
@@ -47,7 +47,7 @@ SET(fann_LIB_SRCS
 floatfann.c
 )
 
-ADD_LIBRARY(fann SHARED ${fann_LIB_SRCS})
+ADD_LIBRARY(fann STATIC ${fann_LIB_SRCS})
 
 SET_TARGET_PROPERTIES(fann PROPERTIES VERSION ${VERSION} SOVERSION 2 )
 INSTALL(TARGETS fann DESTINATION ${LIB_INSTALL_DIR} )
